make.py

```diff
@@ -0,0 +1,341 @@
+            wf.write(build_idempotent_sql_file(idm_file))
+            wf.write("\n\n\n")
+        wf.flush()
+        wf.close()
+    for prior_version in prior_versions():
+        dest = project_dir().joinpath(fr"sql/ai--{prior_version}--{this_version()}.sql")
+        dest.unlink(missing_ok=True)
+        shutil.copyfile(osf, dest)
+
+
+def clean_sql() -> None:
+    for f in sql_dir().glob(f"ai--*.*.*--{this_version()}.sql"):
+        f.unlink()
+    sql_dir().joinpath(f"ai--{this_version()}.sql").unlink()
+
+
+def extension_dir() -> Path:
+    if shutil.which("pg_config") is None:
+        print(f"pg_config not found", file=sys.stderr)
+        exit(1)
+    proc = subprocess.run(f"pg_config --sharedir", check=True, shell=True, env=os.environ, text=True,
+                          capture_output=True)
+    return Path(str(proc.stdout).strip()).resolve().absolute().joinpath("extension")
+
+
+def install_sql() -> None:
+    ext_dir = extension_dir()
+    if not ext_dir.exists():
+        print(f"extension directory does not exist: {ext_dir}", file=sys.stderr)
+        exit(1)
+    build_sql()
+    control_file = sql_dir().joinpath("ai.control")
+    shutil.copyfile(control_file, ext_dir.joinpath(control_file.name))
+    for src in sql_dir().glob("ai--*.sql"):
+        dest = ext_dir.joinpath(src.name)
+        shutil.copyfile(src, dest)
+
+
+def uninstall_sql() -> None:
+    ext_dir = extension_dir()
+    if not ext_dir.exists():
+        return
+    ext_dir.joinpath("ai.control").unlink(missing_ok=True)
+    for f in ext_dir.glob("ai--*.sql"):
+        f.unlink()
+
+
+def python_install_dir() -> Path:
+    # think long and hard before you change this path
+    # don't do it. i'm warning you
+    # seriously.
+    # you'll wreck old versions. look at build_idempotent_sql_file()
+    return Path(f"/usr/local/lib/pgai").resolve().absolute()  # CONTROLS WHERE THE PYTHON LIB AND DEPS ARE INSTALLED
+
+
+def install_old_py_deps() -> None:
+    # this is necessary for versions prior to 0.4.0
+    # we will deprecate these versions and then get rid of this function
+    old_reqs_file = src_dir().joinpath("old_requirements.txt")
+    if old_reqs_file.exists():
+        env = {k: v for k, v in os.environ.items()}
+        env["PIP_BREAK_SYSTEM_PACKAGES"] = "1"
+        subprocess.run(f"pip install --compile -r {old_reqs_file}", shell=True, check=True, env=env)
+
+
+def install_prior_py() -> None:
+    install_old_py_deps()
+    for version in prior_versions():
+        if version in {"0.3.0", "0.2.0", "0.1.0"}:  # these are handled by install_old_py_deps()
+            continue
+        if os.sep in version:
+            print(f"'{os.sep}' in version {version}. this is not supported", file=sys.stderr)
+            exit(1)
+        version_target_dir = python_install_dir().joinpath(version)
+        if version_target_dir.exists():
+            continue
+        tmp_dir = Path("/tmp/pgai").joinpath(version)
```

Comment:
You can use:

```python
import tempfile
with tempfile.TemporaryDirectory() as dir:
    ...
```

Reply:
I like it! Lemme change that. That's more portable.

Reply:
Any ideas on an appropriate place to put the Python stuff on a Windows machine? `C:\pgai`? I haven't touched Windows in so long...

Reply:
fixed.

Reply:
I think something like

C:\Program Files\pgai
C:\Program Files (x86)\pgai